test_databaseconnect.py

```diff
@@ -0,0 +1,151 @@
+import databaseconnect
+import config
+import pytest
+import os
+import dotenv
+import mysql.connector  as mysql
+
+dotenv.load_dotenv()
+
+class TestClass:
+    
+    @pytest.fixture(scope="session", autouse=True)
+    def setup_init(self):
+        # Will be executed before the first test
+        config.database = "test"
+        try:
+            test_db = mysql.connect(
+                host=config.host,
+                user=config.user,
+                passwd=config.password
+            )
+
+            cursor = test_db.cursor()
+            cursor.execute("CREATE DATABASE {}".format(config.database))
+            print("test database created")
+
+        except Exception:
+            print ("Failed to create test database")
+            # rolling back to orignal db
+            config.database = os.getenv("mapbotdatabase")
```

Comment:
Using `os.getenv()` is not encouraged as it depends on `config.py` and further PRs such as #104 may break the code. Rather just `import config` and USE `config.database`. Changing `config.database` outside of `config.py` for tests may cause inconsistencies.

@vishakha-lall this needs to be looked at.

Reply:
Certainly, these changes need to be addressed

Reply:
But the problem I was facing was that every method of `databaseconnect.py` is invoking `connection_to_database` method to connect with the databse and `connection_to_database` is accesing db name from `config.database`. So, no matter what I right in `setup_init` for testing all operations will be performed on the main database that is specified in `config.py` only rather than on a temporary database.

@chttrjeankr  @vishakha-lall can you give me any suggestion to resolve this issue.

Reply:
Take a look at [this](https://github.com/vishakha-lall/MapBot/blob/gssoc-master/databaseconnect.py#L19) line to see how the current bot connects to the database. I can see [this](https://github.com/vishakha-lall/MapBot/pull/103/files#diff-a446acc8912f08ce19c973bbebea2999R17) line of yours is somewhat different and misses the `database` parameter. Pass `database=database_test` to make it connect to your test database. Follow these resources: [1](https://dev.mysql.com/doc/connector-python/en/connector-python-api-mysql-connector-connect.html) and [2](https://dev.mysql.com/doc/connector-python/en/connector-python-connectargs.html) for understanding the call.

Reply:
First of all we will need to create `test` database manually (as referred [here](https://github.com/vishakha-lall/MapBot/blob/gssoc-master/README.md#how-to-set-me-up)) if we want to connect to any database directly using [this](https://dev.mysql.com/doc/connector-python/en/connector-python-api-mysql-connector-connect.html).
Apart from that, what I was saying was Lets say we have done above thing and passed  `database=database_test` [here](https://github.com/vishakha-lall/MapBot/pull/103/files#diff-a446acc8912f08ce19c973bbebea2999R17), then during the tests, methods like [this one](https://github.com/vishakha-lall/MapBot/blob/eafbd3739fb1235c056a01f13b65cf8ea476d375/databaseconnect.py#L46) will invoke [this](https://github.com/vishakha-lall/MapBot/blob/eafbd3739fb1235c056a01f13b65cf8ea476d375/databaseconnect.py#L47) method and it will eventually connect them to any different db (refer [this](https://github.com/vishakha-lall/MapBot/blob/gssoc-master/databaseconnect.py#L19)) which is specified in `config` and operations will not be done on test database that we are connected to in test file.

So, again no matter what I right as `database=database_test` for testing all operations will be performed on the main database itself that is specified in `config.py` only rather than on a `database_test`.

@chttrjeankr this is the problem

Reply:
@msk4862 I understand the problem you're facing with this issue. Before we discuss how to solve that, I would say that there is no need for this particular test. Unit tests are written to test functionalities implemented by you. Creating a new database is done by simply invoking a function from a pre-existing module which doesn't require testing from our end.
@vishakha-lall @chttrjeankr

Reply:
@janakrajchadha makes sense. @msk4862 let's omit this for now

Reply:
@janakrajchadha  actually that's not a test, I'm just creating a temporary database (as [suggested](https://github.com/vishakha-lall/MapBot/issues/74#issuecomment-602421898)), So that during the test all the changes are performed on this temporary database rather than on the main database, and I'm deleting this temporary db after the tests.